STAN.CLIENT/AsyncSubscription.cs

```diff
@@ -118,21 +117,36 @@ internal void subscribe(string subRequestSubject, string subject, string qgroup,
         public void Unsubscribe()
         {
             string linbox = null;
+            string lAckInbox = null;
             Connection lsc = null;
 
-            lock (mu)
+            rwLock.EnterWriteLock();
+
+            try
             {
                 if (sc == null)
                     throw new StanBadSubscriptionException();
 
                 lsc = sc;
                 sc = null;
 
+                linbox = inboxSub.Subject;
                 inboxSub.Unsubscribe();
-                linbox = inbox;
+                inboxSub = null;
+
+                lAckInbox = ackInbox;
+                ackInbox = null;
+            }
+            catch

```

Comment:
Since the there is no logic in here you can elide the `catch { throw; }` safely.